### PR TITLE
Add change tracking guide

### LINKED ISSUE #222: Document new change tracking methods

### PR DESCRIPTION
**Pull request by paulcsmith**
Closes #222

### CODE REVIEW

File: src/actions/guides/database/validating-saving.cr [hunk 1]
```diff
@@ -436,6 +437,60 @@ class Guides::Database::ValidatingSaving < GuideAction
...
+
+    The following change tracking methods are available:
+
+    * `changed?` - returns `true` if the attribute value has changed.
+    * `changed?(from: value)` - returns `true` if the attribute has changed
+      from the passed in value to anything else.
+    * `changed?(to: value)` - returns `true` if the attribute value has changed
+      to the passed in value.
+    * `original_value` - returns the original value before it was changed. If the
+      attribute is unchanged `value` and `original_value` will be the same.
```
Comment (jwoertink) [inline src/actions/guides/database/validating-saving.cr]:
```suggestion
      attribute is unchanged, `value` and `original_value` will be the same.
```

File: src/actions/guides/database/validating-saving.cr [hunk 2]
```diff
@@ -436,6 +437,60 @@ class Guides::Database::ValidatingSaving < GuideAction
...
+
+    > You can also combine `from` and `to` together: `name.changed?(from: nil, to: "Joe")`
```
Comment (jwoertink) [inline src/actions/guides/database/validating-saving.cr:458]: oh, that's sweet!

File: src/actions/guides/database/validating-saving.cr [hunk 3]
```diff
@@ -436,6 +437,60 @@ class Guides::Database::ValidatingSaving < GuideAction
...
+      permit_columns name, email, admin
+
+      before_save do
+        if admin.changed?(to: true)
+          validate_company_email
+        end
+      end
+
+      def validate_company_email
+        if !email.ends_with?("@my-company.com")
```
Comment (jwoertink) [inline src/actions/guides/database/validating-saving.cr]: Not totally sure, but would this actually be `!email.value.ends_with?...`?

File: src/actions/guides/database/validating-saving.cr [hunk 4]
```diff
@@ -436,6 +437,60 @@ class Guides::Database::ValidatingSaving < GuideAction
...
+          email.add_error("must be from @my-company.com to be an admin")
+        end
+      end
+
+      after_save :log_changes
```
Comment (jwoertink) [inline src/actions/guides/database/validating-saving.cr]:
```suggestion
      after_save log_changes
```

You might be able to do a symbol here, but other docs don't use it that way, so for consistency

File: src/posts/lucky_0.19.cr [hunk 5]
```diff
@@ -50,6 +50,9 @@ class Lucky019Release < BasePost
     end
     ```
 
+    View more details in the
+    [change tracking guide]#{Guides::Database::ValidatingSaving.path(anchor: Guides::Database::ValidatingSaving::ANCHOR_CHANGE_TRACKING)}).
```
Comment (jwoertink) [inline src/posts/lucky_0.19.cr:54]: 👍

**Review by jwoertink — COMMENTED** [review]
Sweet! Just a few small things, but looks good.

File: src/actions/guides/database/validating-saving.cr (hunk 3 above)
Comment (paulcsmith) [reply src/actions/guides/database/validating-saving.cr]: Nice catch!

**Comment by paulcsmith** [comment]
Thank you for the feedback!

**Pull request merged**